src/sdk/namespace_blob.js

```diff
@@ -48,54 +51,61 @@ class NamespaceBlob {
             inspect(params)
         );
 
-        const options = {
-            delimiter: params.delimiter,
-            maxResults: params.limit === 0 ? 1 : params.limit
-        };
-        const request_token = params.key_marker && {
-            targetLocation: 0,
-            nextMarker: params.key_marker
-        };
+        const account_name = this.account_name;
+        const hostname = `${account_name}.blob.core.windows.net`;
+        const delimiter = params.delimiter || '/';
+        const maxResults = params.limit === 0 ? 1 : params.limit;
 
-        // TODO the sdk forces us to send two separate requests for blobs and dirs
-        // although the api returns both, so we might want to bypass the sdk
-        const [blobs, dirs] = await P.join(
-            P.fromCallback(callback =>
-                this.blob.listBlobsSegmentedWithPrefix(
-                    this.container,
-                    params.prefix || null,
-                    request_token,
-                    options,
-                    callback)
-            ),
-            P.fromCallback(callback =>
-                this.blob.listBlobDirectoriesSegmentedWithPrefix(
-                    this.container,
-                    params.prefix || null,
-                    request_token,
-                    options,
-                    callback)
-            )
-        );
+        let path = `https://${account_name}.blob.core.windows.net/${this.container}?restype=container&comp=list&delimiter=${delimiter}&maxresults=${maxResults}`;
```

Comment:
OK so you just have to add some comments to explain what we're doing here and why.
That the Azure storage SDK does not allow it, and provide a reference to the SDK code, etc.
But this is a pile of code which is un-parseable without some comments in free language.
Also I would highly recommend to write this code in a utility module rather than inline in namespace_blob.

Reply:
moved to blob_utils and added comment :)